Jenkinsfile

```diff
@@ -0,0 +1,142 @@
+                            sh """
+                                curl --silent --show-error \
+                                    -H "Content-Type: application/json" \
+                                    -H "Accept: application/vnd.github.v3+json" \
+                                    -H "Authorization: token \${GH_TOKEN}" \
+                                    -d '{"tag_name":"'"v${VERSION}"'", "generate_release_notes": true, "name": "'"v${VERSION} - \$(date +"%Y.%m.%d")"'"}' \
+                                    https://api.github.com/repos/rapid7/recog/releases > recog-content-releases-response.json
+                            """
+                        }
+                    }
+                }
+
+                stage('Process release response') {
+                    steps {
+                        script {
+                            // Note: Inspect the response for error message since curl exits with code 0
+                            // for the "Validation failed" response with status code 422 Unprocessable Entity
+                            RELEASE_ERROR_MSG=sh( script: 'cat recog-content-releases-response.json | jq -r .message', returnStdout: true).trim()
+                            echo "[DEBUG] RELEASE_ERROR_MSG = ${RELEASE_ERROR_MSG}"
+                            
+                            if (RELEASE_ERROR_MSG != 'null') {
+                                echo 'Failed to create release.'
+                                sh 'cat recog-content-releases-response.json'
+                                currentBuild.result = 'FAILURE'
+                                return
+                            }
+
+                            echo "[DEBUG] Create release response:"
+                            sh 'cat recog-content-releases-response.json'
+                            echo "[DEBUG] upload_url:"
+                            sh 'cat recog-content-releases-response.json | jq .upload_url'
+                            echo "[DEBUG] processed upload_url:"
+                            sh 'cat recog-content-releases-response.json | jq -r .upload_url | cut -f 1 -d "{"'
+
+                            RELEASE_HTML_URL=sh(script: 'cat recog-content-releases-response.json | jq -r .html_url', returnStdout: true).trim()
+                            echo "[*] Release v${VERSION} available at: ${RELEASE_HTML_URL}"
+
+                            // Note: This returns an `upload_url` key corresponding to the endpoint for uploading release assets.
+                            // This key is a [hypermedia resource](https://docs.github.com/rest/overview/resources-in-the-rest-api#hypermedia).
+                            // All URLs are expected to be proper RFC 6570 URI templates.
+                            UPLOAD_URL=sh(script: 'cat recog-content-releases-response.json | jq -r .upload_url | cut -f 1 -d "{"', returnStdout: true).trim()
+                        }
+                    }
+                }
```

Comment:
Tried to keep this consistent with the original job - including most of the debug lines. I was considering simplifying this - particularly it shouldn't be necessary if we combine the release requests. Though can definitely keep them if it may be helpful.